controller/slot_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require_relative '../model/slot'
+require_relative '../model/car'
+require_relative '../views/slot_view'
+
+# Controller Class for Slot
+class SlotController
+
+  DEFAULT_LOT_SIZE = 10
+  def initialize(size = DEFAULT_LOT_SIZE)
+    return unless Slot.count.zero?
+
+    size.times do
+      Slot.create
+    end
+  end
+
+  def park_car(registration_number)
+    car_id = (Car.where(registration_number:).first || Car.create(registration_number:))[:id]
```

Comment:
Finding a car by registration number or creating a new car to get the car object can be methods on the car model instead of doing the actions in the controller. From the car object, id can be identified.